packages/taiko-client/prover/proof_submitter/proof_submitter.go

```diff
@@ -126,20 +128,42 @@ func (s *ProofSubmitter) RequestProof(ctx context.Context, event *bindings.Taiko
 	}
 
 	// Send the generated proof.
-	result, err := s.proofProducer.RequestProof(
-		ctx,
-		opts,
-		event.BlockId,
-		&event.Meta,
-		header,
-	)
-	if err != nil {
-		return fmt.Errorf("failed to request proof (id: %d): %w", event.BlockId, err)
+	var result *proofProducer.ProofWithHeader
+	if err := backoff.Retry(
+		func() error {
+			// Check if there is a need to generate proof
+			proofStatus, err := rpc.GetBlockProofStatus(
+				ctx,
+				s.rpc,
+				opts.BlockID,
+				opts.ProverAddress,
+				s.proverSetAddress,
+			)
+			if err != nil {
+				return err
+			}
+			if proofStatus.IsSubmitted && !proofStatus.Invalid {
+				return nil
+			}
+
+			if result, err = s.proofProducer.RequestProof(
+				ctx,
+				opts,
+				event.BlockId,
+				&event.Meta,
+				header,
+			); err != nil {
+				return fmt.Errorf("failed to request proof (id: %d): %w", event.BlockId, err)
+			}
+			return nil
+		},
+		backoff.WithContext(backoff.NewConstantBackOff(proofPollingInterval), ctx),
+	); err != nil {
+		log.Error("Request proof error", "error", err)
```

Comment:
Shall we `return` here? Otherwise we will send an empty `result` to `s.resultCh`?

Reply:
Oh, yes, when `proofStatus` is valid, we will send empty.
This doesn't have errors in the processing flow, but it does cause errors in the statistics.
Let me fix it.